clang/include/clang/Tooling/DependencyScanning/DependencyScanningCASFilesystem.h

```diff
@@ -63,6 +63,7 @@ class DependencyScanningCASFilesystem : public llvm::cas::ThreadSafeFileSystem {
   createThreadSafeProxyFS() override;
 
   llvm::ErrorOr<llvm::vfs::Status> status(const Twine &Path) override;
+  bool exists(const Twine &Path) override;
```

Comment:
It don't really like this API for skipping caching for the lookup (also the comments should be here, instead of in the implementation).

This kind of encourages skipping tracking anti dependency. I would rather a more specific approach that depscan should skip autolink check.

Reply:
> This kind of encourages skipping tracking anti dependency.

How so? The dependency is still tracked by `CachingOnDiskFileSystem`.

> I would rather a more specific approach that depscan should skip autolink check.

How would we do this without breaking framework autolinking?

Reply:
> also the comments should be here, instead of in the implementation

I tweaked the comment to clarify the caching fs still tracks the access. I think the comment is in the right place though, because it's about why the implementation can forward directly to the caching fs, not about the behaviour the client will see on the `exists` call itself.